### DISCUSSION TITLE
nvim-lint and eslint

### DISCUSSION THREAD

**Discussion by eriklindebratt** [LazyVim/LazyVim#1741] [Q&A] [answered]
Hi! 👋
I'm confused as to the role of `nvim-lint`, especially when it comes to ESLint and the `lazyvim.plugins.extras.linting.eslint` extra.

Except from adding the `lazyvim.plugins.extras.linting.eslint` extra and installing the `eslint-lsp` client using Mason (which is done automatically) I don't seem to be needing anything to both see linting errors in my code and being able to correct them using `:EslintFixAll`. This is awesome! But I can't help but wonder:

1. I don't seem to need to add eslint to `nvim-lint`'s `linters_by_ft`. Why? When inspecting `require('lint').linters_by_ft` I only get linters for `dockerfile`, `fish`, `terraform` and `tf`.
2. What's the role of the `lazyvim.plugins.extras.linting.eslint` and what does it do, as compared to `nvim-lint`?
3. Why don't I need the something similar to `lazyvim.plugins.extras.linting.eslint` for other linters added to `nvim-lint`'s `linters_by_ft`?

Thanks for all the great work!

**Comment by folke** [comment, marked as answer]
the eslint extra is an eslint LSP server, so you don't need the eslint linter as well through nvim-lint.

The eslint LSP provides more features and is faster than running cmdline eslint through nvim-lint

**Reply by eriklindebratt** [reply]
Boy that was fast, thanks! Got it.
